tests/test_transform/factory/test_correctness.py

```
@@ -146,6 +146,96 @@ def test_correctness_attention_heads(type, randomize, head_dim, input_batch_size
+    """Test that WEIGHT_OUTPUT transforms correctly handle bias.
+
+    For WEIGHT_OUTPUT on a Linear with bias:
+        y' = R @ (W @ x + b) = (R @ W) @ x + (R @ b)
+    Both weight and bias must be transformed.
+    """
+    size = (4, 8)
+    module = torch.nn.Linear(*size, bias=True)
+    scheme = TransformScheme(type=type, randomize=randomize, head_dim=head_dim)
+    factory = TransformFactory.from_scheme(scheme, name="")
+
+    input = torch.rand((input_batch_size, 5, size[0]))
```

Ruff (0.15.9)

[error] Variable `input` is shadowing a Python builtin

(A001)